Riot/ViewController/GroupRoomsViewController.m

```diff
@@ -0,0 +1,602 @@
+        [self.mxSession updateGroupRooms:_group success:nil failure:^(NSError *error) {
+            
+            NSLog(@"[GroupRoomsViewController] viewWillAppear: group rooms update failed %@", _group.groupId);
+            
+        }];
+    }
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [super viewWillDisappear:animated];
+    
+    // cancel any pending search
+    [self searchBarCancelButtonClicked:_searchBarView];
+}
+
+- (void)withdrawViewControllerAnimated:(BOOL)animated completion:(void (^)(void))completion
+{
+    // Check whether the current view controller is displayed inside a segmented view controller in order to withdraw the right item
+    if (self.parentViewController && [self.parentViewController isKindOfClass:SegmentedViewController.class])
+    {
+        [((SegmentedViewController*)self.parentViewController) withdrawViewControllerAnimated:animated completion:completion];
+    }
+    else
+    {
+        [super withdrawViewControllerAnimated:animated completion:completion];
+    }
+}
+
+- (void)setGroup:(MXGroup*)group withMatrixSession:(MXSession*)mxSession
+{
+    // Cancel any pending search
+    [self searchBarCancelButtonClicked:_searchBarView];
+    
+    _mxSession = mxSession;
+    
+    [self addMatrixSession:mxSession];
+    
+    [self refreshDisplayWithGroup:group];
+}
+
+#pragma mark -
+
+- (void)registerOnGroupChangeNotifications
+{
+    [self cancelRegistrationOnGroupChangeNotifications];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didUpdateGroupRooms:) name:kMXSessionDidUpdateGroupRoomsNotification object:self.mxSession];
+}
+
+- (void)cancelRegistrationOnGroupChangeNotifications
+{
+    // Remove any pending observers
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)didUpdateGroupRooms:(NSNotification *)notif
+{
+    // Update here the displayed group instance with the one stored in the session (if any).
+    MXGroup *group = [self.mxSession groupWithGroupId:_group.groupId];
+    
+    [self refreshDisplayWithGroup:(group ? group : _group)];
+}
+
+- (void)refreshDisplayWithGroup:(MXGroup *)group
+{
+    _group = group;
+    
+    if (_group)
+    {
+        _searchBarHeader.hidden = NO;
+        
+        [self registerOnGroupChangeNotifications];
+        
+        groupRooms = _group.rooms.chunk;
+    }
+    else
+    {
+        // Search bar header is hidden when no group is provided
+        _searchBarHeader.hidden = YES;
+        
+        [self cancelRegistrationOnGroupChangeNotifications];
+        
+        groupRooms = nil;
+    }
+    
+    // Reload search result if any
+    if (currentSearchText.length)
+    {
+        NSString *searchText = currentSearchText;
+        currentSearchText = nil;
+        
+        [self searchBar:_searchBarView textDidChange:searchText];
+    }
+    else
+    {
+        [self refreshTableView];
+    }
+}
+
+- (void)startActivityIndicator
```

Comment:
The activity indicator is not used for the moment (this is just a notice)

Reply:
agree, I kept it because it will be used when we will enable the admin to remove rooms